naga/src/front/spv/mod.rs

```diff
@@ -3944,7 +3961,68 @@ impl<I: Iterator<Item = u32>> Frontend<I> {
                     );
                     emitter.start(ctx.expressions);
                 }
-                _ => return Err(Error::UnsupportedInstruction(self.state, inst.op)),
+                Op::AtomicIIncrement => {
+                    inst.expect(6)?;
+                    let start = self.data_offset;
+                    let result_type_id = self.next()?;
+                    let result_id = self.next()?;
+                    let pointer_id = self.next()?;
+                    let scope_id = self.next()?;
+                    let memory_semantics_id = self.next()?;
+                    // Store the op for a later pass where we "upgrade" the pointer type
+                    let atomic = AtomicOp {
+                        instruction: AtomicOpInst::AtomicIIncrement,
+                        result_type_id,
+                        result_id,
+                        pointer_id,
+                        scope_id,
+                        memory_semantics_id,
+                    };
+                    self.lookup_atomic.insert(pointer_id, atomic);
+
+                    log::trace!("\t\t\tlooking up expr {:?}", pointer_id);
+                    let (p_lexp_handle, p_lexp_ty_id) = {
+                        let lexp = self.lookup_expression.lookup(pointer_id)?;
+                        (lexp.handle, lexp.type_id)
+                    };
+                    log::trace!("\t\t\tlooking up type {:?}", pointer_id);
+                    let p_ty = self.lookup_type.lookup(p_lexp_ty_id)?;
+
+                    // Create an expression for our result
+                    let r_lexp_handle = {
+                        let expr = crate::Expression::AtomicResult {
+                            ty: p_ty.handle,
+                            comparison: false,
+                        };
+                        let handle = ctx.expressions.append(expr, self.span_from_with_op(start));
+                        self.lookup_expression.insert(
+                            result_id,
+                            LookupExpression {
+                                handle,
+                                type_id: result_type_id,
+                                block_id,
+                            },
+                        );
+                        handle
+                    };
+
+                    // Create a literal "1" since WGSL lacks an increment operation
+                    // Create a statement for the op itself
+                    let one_lexp_handle = {
+                        let expr = crate::Expression::Literal(crate::Literal::I32(1));
+                        ctx.expressions.append(expr, self.span_from_with_op(start))
+                    };
```

Comment:
Isn't it possible for the base type to be either `i32` or `u32`? Would it be possible to use `make_index_literal` here? That might help match the `crate::Literal` variant with the atomic's operand type.

Reply:
Yup! You're right. It has to match the type of `pointer_id`.